script.js

```diff
@@ -0,0 +1,211 @@
+
+  toggleKeyClass(pressedKey, true);
+
+  if (pressedKey === 'shift') {
+    toggleKeyClass('shift', true);
+  }
+
+  press(event.key);
+
+  if (pressedKey === 'enter') {
+    press('\n');
+  }
+
+  if (isSpaceKey) {
+    party.wordCounter++;
+  }
+
+  if (
+    party.currentStringIndex === party.strings.length - 1 &&
+    party.currentPressedIndex === party.strings[party.currentStringIndex].length
+  ) {
+    input.blur();
+    return;
+  }
+}
+
+function keyUpHandler(event) {
+  const isSpaceKey = event.key === ' ';
+  const pressedKey = isSpaceKey ? 'space' : event.key.toLowerCase();
+
+  toggleKeyClass(pressedKey, false);
+
+  if (pressedKey === 'shift') {
+    toggleKeyClass('shift', false);
+  }
+}
+
+function createParty(text) {
+  const party = {
+    text,
+    strings: [],
+    maxStringLength: 70,
+    maxShowStrings: 5,
+    currentStringIndex: 0,
+    currentPressedIndex: 0,
+    errors: [],
+    started: false,
+    startTimer: 0,
+    timerCounter: 0,
+    symbolCounter: 0,
+    wordCounter: 0,
+    errorCounter: 0,
+    statisticFlag: false,
+  };
+  party.text = party.text.replace(/\n/g, '\n ');
+  const words = party.text.split(' ');
+  let string = [];
+
+  for (const word of words) {
+    const newStringLength = [...string, word].join(' ').length + !word.includes('\n');
+
+    if (newStringLength > party.maxStringLength) {
+      party.strings.push(string.join(' ') + ' ');
+      string = [];
+    }
+
+    string.push(word);
+
+    if (word.includes('\n')) {
+      party.strings.push(string.join(' '));
+    }
+  }
+  if (string.length) {
+    party.strings.push(string.join(' '));
+  }
+  return party;
+}
+
+function press(pressedKey) {
+  party.started = true;
+  if (!party.statisticFlag) {
+    party.statisticFlag = true;
+    party.startTimer = Date.now();
+  }
+  const string = party.strings[party.currentStringIndex];
+  const mustKey = string[party.currentPressedIndex];
+  if (pressedKey === mustKey) {
+    party.currentPressedIndex++;
+    if (
+      string.length <= party.currentPressedIndex &&
+      party.currentStringIndex < party.strings.length - 1
+    ) {
+      party.currentPressedIndex = 0;
+      party.currentStringIndex++;
+      party.statisticFlag = false;
+    }
+  } else if (!party.errors.includes(mustKey)) {
+    party.errors.push(mustKey);
+    party.errorCounter++;
+  }
+  party.symbolCounter++;
+  party.timerCounter = Date.now() - party.startTimer;
+  viewUpdate(party, textExample);
+}
+
+function statisticCount() {
+  if (party.started) {
+    symbolsPerMinute.textContent = Math.round((60000 * party.symbolCounter) / party.timerCounter);
+    errorPercent.textContent =
+      Math.floor((10000 * party.errorCounter) / party.symbolCounter / 100) + '%';
+    wordsPerMinute.textContent = Math.round((60000 * party.wordCounter) / party.timerCounter);
+  }
+}
+
+function viewUpdate(party,textExample) {
+  const string = party.strings[party.currentStringIndex];
```

Comment:
const string = party.strings[party.currentStringIndex];
const shownString = party.strings.slice(...)
const firstLine = document.createElement('div');
let div = document.createElement('div');
const doneLine = document.createElement('span');

prefer use a object for this variables.